### DISCUSSION TITLE
The `identify.exe -ping` command fails when supplied with globbing pattern

### DISCUSSION THREAD

**Discussion by yansklyarenko** [ImageMagick/ImageMagick#2239] [Help] [answered]
I'm observing a strange behavior of the `identify.exe` utility (Windows 10).
Let's say I have a single JPG file in a directory. If that file has a usual name (for example, `image.jpg`), the following command works just fine:
`C:\ProgramData\chocolatey\bin\identify.exe -ping *.jpg`
and produces expected output:
`image.jpg JPEG 318x313 318x313+0+0 8-bit sRGB 12853B 0.000u 0:00.000`

However, as soon as the file name starts with `.` - the same command fails with the following error:
`identify.exe: UnableToOpenBlob '*.jpg': Invalid argument @ error/blob.c/OpenBlob/3497.`

It's interesting that if you supply the exact file name to the command, it works just fine:
`C:\ProgramData\chocolatey\bin\identify.exe -ping .starts_with_dot.jpg`
outputs:
`.starts_with_dot.jpg JPEG 318x313 318x313+0+0 8-bit sRGB 12853B 0.000u 0:00.000`

Is it expected? Is it possible to work this around by providing some parameters to the identify.exe?

Thanks in advance!

**Comment by dlemstra** [comment]
Thanks for reporting this. Will take a look at it tonight and check why this happens.

**Reply by yansklyarenko** [reply]
Thanks a lot, @dlemstra!

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by yansklyarenko** [reply]
Thanks a lot! I can see that the new beta binaries are there, however, the 7z for Windows seems to be empty: https://www.imagemagick.org/download/beta/ImageMagick-7.0.10-24~beta20200707-windows.7z. Is it expected? What's the right way to build the tools on Windows 10 to try things out?

**Reply by dlemstra** [reply]
We don't have beta builds of the Windows release yet on a daily basis. Will need to investigate on how to make that possible with our installer. For now you could download a "pre-release" from this page: https://github.com/ImageMagick/ImageMagick/actions?query=workflow%3Arelease.

**Reply by yansklyarenko** [reply]
Great! I've just verified the fix, and it works with 7.0.10-24! Amazing job, people! Thank you!
